Sources/FoundationExtensions/URL.swift

```diff
@@ -6,4 +6,17 @@ extension URL {
         o.appendPathExtension(ext)
         return o
     }
+    
+    public subscript(queryParam: String) -> String? {
+        guard let components = URLComponents(url: self, resolvingAgainstBaseURL: false)
```

Comment:
This subscript would be very convenient, but it would also come at the cost of performance and the api loss since `resolvingAgainstBaseURL` is not available and each call creates new components object.

This could be addressed by adding such a subscript to the URLComponents type instead. I suspect there would be many more extensions we'd like to add to this type in the future. As for the URL itself, you can extend it only to provide more convenient ways of creating URLComponents from it - e.g. a method with `resolvingAgainstBaseURL` parameter defaulting to `false`... This allows the user to reuse the components object.